pgrx/src/memcx.rs

```diff
@@ -16,27 +18,35 @@ pub struct MemCx<'mcx> {
 }
 
 impl<'mcx> MemCx<'mcx> {
-    /// You probably shouldn't be using this.
+    /// Wrap the provided [`pg_sys::MemoryContext`]
+    ///
+    /// # Safety:
+    /// Assumes the provided [`pg_sys::MemoryContext`] is valid and properly initialized.
+    /// This method does check to ensure the pointer is non-null, but that is the only sanity
+    /// check that is performed.
     pub(crate) unsafe fn from_ptr(raw: pg_sys::MemoryContext) -> MemCx<'mcx> {
         let ptr = NonNull::new(raw).expect("memory context must be non-null");
         MemCx { ptr, _marker: PhantomData }
     }
 
-    /// You probably shouldn't be using this.
-    pub(crate) unsafe fn alloc_bytes(&self, size: usize) -> *mut u8 {
+    /// Allocate a raw byte buffer `size` bytes in length
+    /// and returns a pointer to the new allocation.
+    pub unsafe fn alloc_bytes(&self, size: usize) -> *mut u8 {
```

Comment:
Needs the safety constraint documented.

Reply:
Still pondering if this should be safe or not...

Reply:
Per our conversation, this appears to be safe. I've made it a safe method.